metadata/errors.go

```diff
@@ -1,22 +1,117 @@
 package metadata
 
-import "github.com/pkg/errors"
+type metadataExistsErr struct {
+	desc string
```

Comment:
Would it make more sense to leave these is as a `cause` and have these error types implement cause?